File: TopCoder04/PHP/02/Main.php
<?php
    class InterestingDigits {
    	public function digits($base)
		{
			$ans = array();
			
			for($i = 2 ; $i< $base; $i++){
				if(($base-1) % $i == 0){
					$ans[] = $i;
				}
			}
			
			return $ans;
		}
    }
	
	$base = 10;
	$id = new InterestingDigits;
	$result = $id->digits($base);
	
	foreach ($result as $value) {
		echo $value ," ";
	}
	
?>

File: TopCoder04/Ruby/02/Main.rb
class InterestingDigits
  def dogots(base)
    ans = []
    for i in 2...base
      if (base-1) % i == 0
        ans << i
      end
    end
    return ans 
  end 
end

puts InterestingDigits.new.dogots(10)
